### DISCUSSION TITLE
runtime behaviour

### DISCUSSION THREAD

**Discussion by Henny022** [profi200/open_agb_firm#97] [Q&A] [answered]
From my understanding, this should behave exactly the same as the official releases from ie the ambassador program, in terms of speed and saving latency etc.?

**Comment by profi200** [comment, marked as answer]
There are [2 timings](https://github.com/profi200/libn3ds/blob/master/source/arm9/drivers/lgy.c#L90-L91) as far as i know. They are pretty close to original hardware timings but certainly not perfect. These have been taken from official ambassador releases.

**Reply by Henny022** [reply]
As long as that's the same as the ambassador releases, that's good.
the games framerate should also be exactly the same since they use the same core and it's independent from the display framerate/whatever open_agb_firm does, if I'm reading everything correctly?

**Reply by profi200** [reply]
Frame rate is the same as the ambassador releases. It all runs in hardware. It's running exactly like it would on a Nintendo DS/DS lite.

On DS it occasionally had to duplicate a frame because GBA and DS refresh rate are ever so slightly different. On 3DS this problem is worked around by altering LCD timings so it stays in sync.

**Reply by Henny022** [reply]
perfect thanks
